tools/k8s-label-visualizer/dot_graph.py

```diff
@@ -0,0 +1,101 @@
+from graphviz import Digraph
+
+import configuration
+
+KUBERNETES_IO_COMPONENT = "app.kubernetes.io/component"
+KUBERNETES_IO_MANAGED_BY = "app.kubernetes.io/managed-by"
+
+
+def get_labels(i):
+    if "metadata" in i and "labels" in i["metadata"]:
+        return i["metadata"]["labels"]
+    return []
+
+
+class GraphGenerator(object):
+
+    def __init__(self, conf: configuration.Configuration, objects, outdir):
+        self.conf = conf
+        self.objects = objects
+        self.outdir = outdir
+
+    def generate(self):
+        self.generate_component_graph()
+        self.generate_managed_by_graph()
+
+    def generate_managed_by_graph(self):
+        dot = Digraph(
+            comment='Managed By', graph_attr={'nodesep': '.5', 'ranksep': '5'}
+        )
+
+        for i in self.objects:
+            node_name = self.get_node_name(i)
+            dot.node(node_name, node_name)
+
+            labels = get_labels(i)
+            for label in labels:
+                if label == KUBERNETES_IO_MANAGED_BY:
+                    managed_node_name = self.transform_alias(
+                        labels[KUBERNETES_IO_MANAGED_BY]
+                    )
+                    dot.node(managed_node_name, managed_node_name)
+                    dot.edge(managed_node_name, node_name)
```

Comment:
Instead of looping over a dictionary to get a key, maybe we can just use random access to this key?

Reply:
done